system/jlib/jevent.cpp

```diff
@@ -838,17 +854,297 @@ EventRecorder eventRecorder;
 
 //---------------------------------------------------------------------------------------------------------------------
 
+class CEventAttribute : public IEventAttribute
+{
+public: // IEventAttribute
+    virtual EventAttr queryId() const override
+    {
+        return id;
+    }
+
+    virtual bool isText() const override
```

Comment:
Ideally this would be pre-calculated (using constexpr functions) and stored in the attribute information.